spec/unit/part_spec.rb

```diff
@@ -4,6 +4,40 @@
   end
 end
 
+RSpec.shared_examples '#new' do
+  context 'Same part class' do
+    let(:new_part) { part.new(name: :user_admin, value: 'ADMIN') }
+
+    it 'returns a new instance of the same class but same context and renderer' do
+      expect(part.class).to eq new_part.class
+      expect(part.context).to eq new_part.context
+      expect(part._renderer).to eq new_part._renderer
+    end
+
+    it 'returns a new instance with different values' do
+      expect(part.class).to eq new_part.class
+      expect(part.value).to_not eq new_part.value
+      expect(part._name).to_not eq new_part._name
+    end
+  end
+
+  context 'Different part class' do
+    before do
+      module Test
+        class AdminPart < Dry::View::Part; end
+      end
+    end
+
+    let(:new_part) { part.new(Test::AdminPart, name: :user_admin, value: 'ADMIN') }
+
+    it 'returns a new class instance but same context and renderer' do
+      expect(part.class).to_not eq new_part.class
+      expect(part.context).to eq new_part.context
+      expect(part._renderer).to eq new_part._renderer
+    end
+  end
+end
```

Comment:
It would be good to tweak this spec so that instead of checking object's state (`context` and esp `_renderer`), it's checking behavior. ie see if rendering works (because it relies on context + renderer).

Also small style problem: `to eq new_part.class` => `to eql(new_part.class)`

Reply:
Great advice @solnic I will tweak the test! 🌶